### PR TITLE
Detect M1 mac and skip JxBrowser

### PR DESCRIPTION
**Pull request by helin24**
If a user is using an M1 mac and the version of IntelliJ native to this chip, then JxBrowser is not currently supported. If we detect this combination then show users the option to open DevTools in the browser instead.

It seems that automatically disabling embedded browser for situations where JxBrowser is not supported results in the setting being somewhat sticky once it is supported, e.g. it seems previously disabling for Big Sur resulted in a much higher proportion of users staying disabled despite no additional errors that would suggest a poorer user experience (maybe this setting sticks around as long as users keep their version of IntelliJ? I'm not sure). So I'm inclined to try prompting users to open in the browser instead. This should be a smaller proportion of users anyway.

Fixes https://github.com/flutter/flutter-intellij/issues/5472

### CODE REVIEW

File: src/io/flutter/view/FlutterView.java [hunk 1]
```diff
@@ -623,12 +627,12 @@ protected void presentClickableLabel(ToolWindow toolWindow, List<LabelInput> lab
...
     for (LabelInput input : labels) {
       if (input.listener == null) {
-        final JLabel descriptionLabel = new JLabel(input.text);
+        final JLabel descriptionLabel = new JLabel("<html>" + input.text + "</html>");
         descriptionLabel.setBorder(JBUI.Borders.empty(5));
         descriptionLabel.setHorizontalAlignment(SwingConstants.CENTER);
         panel.add(descriptionLabel, BorderLayout.NORTH);
       } else {
-        final LinkLabel<String> linkLabel = new LinkLabel<>(input.text, null);
+        final LinkLabel<String> linkLabel = new LinkLabel<>("<html>" + input.text + "</html>", null);
```
Comment (jacob314) [inline src/io/flutter/view/FlutterView.java:635]: adding <html> seems odd. Are they needed? this is still far from valid html with <html> tags added.

File: tool/plugin/lib/edit.dart [hunk 2]
```diff
@@ -136,6 +136,18 @@ List<EditCommand> editCommands = [
     replacement: 'urlClassLoader.addFiles(paths)',
     versions: ['2021.1'],
   ),
+  Subst(
```
Comment (jacob314) [inline tool/plugin/lib/edit.dart:139]: @stevemessick is this the best way to do this?
The alternative I can think of is to use reflection to dynamically call the api.

**Review by jacob314 — COMMENTED** [review]
We should also understand why we accidentally set the sticky opt out instead of making it optional. I expect we changed the value of the setting when we meant to leave it alone but change the value surfaced to users.

File: tool/plugin/lib/edit.dart (hunk 2 above)
Comment (stevemessick) [reply tool/plugin/lib/edit.dart:139]: This allows us to avoid reflection.

**Review by stevemessick — APPROVED** [review]

File: src/io/flutter/view/FlutterView.java (hunk 1 above)
Comment (helin24) [reply src/io/flutter/view/FlutterView.java:635]: I added this to get text wrapping within the panel. I read that we can also get text wrapping by changing to a text area with some other adjustments, but this seems more straightforward.

**Comment by helin24** [comment]
> We should also understand why we accidentally set the sticky opt out instead of making it optional. I expect we changed the value of the setting when we meant to leave it alone but change the value surfaced to users.

I think this is what happened:
- A few versions ago, Big Sur users had a disabled "embedded browser" field 
- If a user did not adjust any of their other Flutter settings, the embedded browser field does not get a value
- If a user DID adjust any of their other Flutter settings, the embedded browser field was set to false, since we were returning false if Big Sur
- Then we re-enabled the embedded browser field by default, which failed to enable for the second group of users, since they  had a false setting.

This seems to reproduce for me locally. I should have made this setting line conditional on not big sur: https://github.com/flutter/flutter-intellij/pull/5065/files#diff-56079033e7fb39f494b815b798ad070de67300e721b0658082279962ab024389R306

So to fix this, I'll override a false setting for Big Sur users once but allow users to change it back afterwards.

**Pull request merged**